File: 0506-relative-ranks/0506-relative-ranks.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        priority_queue<pair<int, int>> pq;

        for (int i = 0; i < score.size(); i++) {
            pq.push({score[i], i});
        }

        vector<string> res(score.size());

        int i = 1;

        while (!pq.empty()) {
            auto pair = pq.top();
            pq.pop();

            int idx = pair.second;

            if (i == 1)
                res[idx] = "Gold Medal";
            else if (i == 2)
                res[idx] = "Silver Medal";
            else if (i == 3)
                res[idx] = "Bronze Medal";
            else
                res[idx] = to_string(i);

            ++i;
        }

        return res;
    }
};

File: 0506-relative-ranks/0506-relative-ranks.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
        Comparator<Pair> pairComparator = Comparator.comparingInt(pair -> pair.first);

        PriorityQueue<Pair> pq = new PriorityQueue<>(pairComparator.reversed());
        
        for(int i = 0; i < score.length; i++){
            pq.offer(new Pair(score[i], i));
        }
        
        String[] res = new String[score.length];
        
        int i = 1;
        
        while(!pq.isEmpty()){
            Pair pair = pq.poll();
            
            if(i == 1)
                res[pair.second] = "Gold Medal";
            
            else if(i == 2)
                res[pair.second] = "Silver Medal";
            
            else if(i == 3)
                res[pair.second] = "Bronze Medal";
            
            else
                res[pair.second] = String.valueOf(i);
            
            ++i;
        }
        
        return res;
    }
}

class Pair{
    int first;
    int second;
    
    public Pair(int first, int second) {
        this.first = first;
        this.second = second;
    }
}